assets/js/admin/course-theme-sidebar.js

```diff
@@ -3,49 +3,147 @@
  */
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
 import { __ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { ToggleControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import useCourseMeta from '../../react-hooks/use-course-meta';
+import { name as courseOutlineBlockName } from '../../blocks/course-outline/outline-block/block.json';
+import { name as courseModuleBlockName } from '../../blocks/course-outline/module-block/block.json';
+import { name as courseLessonBlockName } from '../../blocks/course-outline/lesson-block/block.json';
 
 const SENSEI_THEME = 'sensei-theme';
 const WORDPRESS_THEME = 'wordpress-theme';
+const SENSEI_PREVIEW_QUERY = 'sensei_theme_preview';
+
+const canPreview = ( block ) =>
+	block.name === courseLessonBlockName &&
+	block.attributes.id &&
+	! block.attributes.draft;
```

Comment:
Just a question: Have you explored if we could make something to preview draft lessons too?

Reply:
I did now. 😄 f79540784de9520007e2463e0a00e7224b83baef

I added a support for it. But it looks a bit empty. Course navigation sidebar is empty, because it excludes the unpublished lessons. But we can improve it further in future iterations. LMWYT. See video below


https://user-images.githubusercontent.com/2578542/148183215-8354f762-81a6-4100-99bf-9dbb2540b11f.mp4

Reply:
I think we could start with this version and improve in future iterations, as you said. =)